src/app/core/services/page-title/page-title.service.ts

```diff
@@ -0,0 +1,107 @@
+import {Injectable} from '@angular/core';
+import {Title} from '@angular/platform-browser';
+import {Subject, of, Observable} from 'rxjs';
+import {takeUntil, switchMap, tap} from 'rxjs/operators';
+import * as _ from 'lodash';
+
+import {ProjectService, ClusterService, ApiService} from '../../services';
+import {ParamsService, PathParam} from '../../services/params/params.service';
+import {Cluster} from '../../../shared/entity/cluster';
+import {View, getViewDisplayName, ViewDisplayName} from '../../../shared/entity/common';
+import {NodeDeployment} from '../../../shared/entity/node-deployment';
+
+@Injectable()
+export class PageTitleService {
+  projectName: string;
+  clusterName: string;
+  ndName: string;
+  private readonly _unsubscribe = new Subject<void>();
+
+  constructor(
+    private readonly _titleService: Title,
+    private readonly _params: ParamsService,
+    private readonly _projectService: ProjectService,
+    private readonly _clusterService: ClusterService,
+    private readonly _apiService: ApiService
+  ) {}
+
+  setTitle(url: string): void {
+    const viewName = this._getViewName(url.split('/'));
+    this._titleService.setTitle(viewName);
+
+    this._projectService.selectedProject
+      .pipe(tap(project => (this.projectName = project ? project.name : '')))
+      .pipe(switchMap(_ => this._clusterObservable()))
+      .pipe(tap(cluster => (this.clusterName = cluster ? cluster.name : '')))
+      .pipe(switchMap(_ => this._nodeDeploymentObservable()))
+      .pipe(tap(nd => (this.ndName = nd ? nd.name : '')))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(_ => {
+        this._titleService.setTitle(this._generateTitle(viewName));
+      });
+  }
+
+  private _getViewName(urlArray: string[]): string {
+    if (!_.isEmpty(urlArray)) {
+      let viewName = Object.values(View).find(view => view === urlArray[urlArray.length - 1]);
+      if (_.isEmpty(viewName)) {
```

Comment:
We don't really need `isEmpty` check for a simple string here. `!viewName` should be enough to check for empty/undefined string. Nesting `if` is also not needed here. This could be:

```typescript
let viewName: string;
// Here simple array.length > 0 should be enough since viewName is a result of split and split will always return at least empty array.
if(urlArray.length > 0) {
  viewName = Object.values(View).find(view => view === urlArray[urlArray.length - 1]);
}

// We need to check if array length is actually bigger than 1. isEmpty only checks if it's bigger than 0.
if(!viewName && urlArray.length > 1) {
  viewName = Object.values(View).find(view => view === urlArray[urlArray.length - 2]);
}

return viewName ? getViewDisplayName(viewName) : '';
```